src/Microsoft.SqlTools.ServiceLayer/ModelManagement/ModelManagementService.cs

```diff
@@ -0,0 +1,247 @@
+        /// </summary>
+        /// <param name="parameters">Request parameters</param>
+        /// <param name="requestContext">Request Context</param>
+        public async Task HandleVerifyModelTableRequest(VerifyModelTableRequestParams parameters, RequestContext<VerifyModelTableResponseParams> requestContext)
+        {
+            Logger.Write(TraceEventType.Verbose, "HandleVerifyModelTableRequest");
+            VerifyModelTableResponseParams response = new VerifyModelTableResponseParams
+            {
+            };
+
+            await HandleRequest(parameters, response, requestContext, (dbConnection, parameters, response) =>
+            {
+                response.Verified = ModelOperations.VerifyImportTable(dbConnection, parameters);
+                return response;
+            });
+        }
+
+        /// <summary>
+        /// Handles configure model table request
+        /// </summary>
+        /// <param name="parameters">Request parameters</param>
+        /// <param name="requestContext">Request Context</param>
+        public async Task HandleConfigureModelTableRequest(ConfigureModelTableRequestParams parameters, RequestContext<ConfigureModelTableResponseParams> requestContext)
+        {
+            Logger.Write(TraceEventType.Verbose, "HandleConfigureModelTableRequest");
+            ConfigureModelTableResponseParams response = new ConfigureModelTableResponseParams
+            {
+            };
+
+            await HandleRequest(parameters, response, requestContext, (dbConnection, parameters, response) =>
+            {
+                ModelOperations.ConfigureImportTable(dbConnection, parameters);
+                return response;
+            });
+        }
+
+        private async Task HandleRequest<T, TResponse>(
+            T parameters, 
+            TResponse response, 
+            RequestContext<TResponse> requestContext,
+            Func<IDbConnection, T, TResponse, TResponse> operation) where T : ModelRequestBase where TResponse : ModelResponseBase
+        {
+            try
+            {
+                ConnectionInfo connInfo;
+                ConnectionServiceInstance.TryFindConnection(
+                    parameters.OwnerUri,
+                    out connInfo);
+                if (connInfo == null)
+                {
+                    response.ErrorMessage = SR.ConnectionServiceDbErrorDefaultNotConnected(parameters.OwnerUri);
+                    response.Success = false;
+                }
+                else
+                {
+                    response.Success = true;
+                    using (IDbConnection dbConnection = ConnectionService.OpenSqlConnection(connInfo))
+                    {
+                        response = operation(dbConnection, parameters, response);
+                        response.Success = true;
+                    }
+                }
+            }
+            catch (Exception e)
+            {
+                response.ErrorMessage = e.Message;
```

Comment:
You don't want to SendError instead? That'll make it throw on the extension side which is usually what we want to happen so we don't have constantly be checking the response state. (this goes for all the other places you're setting success to false as well)

Reply:
yeah I put it back to send error